theano/sandbox/cuda/basic_ops.py

```diff
@@ -3044,6 +3044,24 @@ def make_node(self, *axis_and_tensors):
         as_tensor_variable_args = [as_cuda_ndarray_variable(x)
                                    for x in tensors]
 
+        # If axis is negative, we need to increment it with the number
+        # of dimensions of the smallest tensor variable
+        axis_int = int(axis.eval())
```

Comment:
Also, use ```X.ndim``` to get number of dimensions rather than ```len(list(X.shape))```. 

To get ```axis``` value, do something like [this](https://github.com/Theano/Theano/blob/master/theano/tensor/basic.py#L3440).